### PR TITLE
feat: Request class

### PR DESCRIPTION
**Pull request by dpiza**
## What
Adds a Request class

## Why
The Request class will be responsible for parsing and storing the information received from the HTTP request

### CODE REVIEW

File: include/Request/Request.hpp [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+#include <cstdlib>
+#include <iostream>
+#include <sstream>
+#include <string>
+
+#include "WebServerException/WebServerException.hpp"
+
+#define TRUE 1
+#define FALSE 0
```
Comment (koukunwlk) [inline include/Request/Request.hpp:11]: Why did you redefine these variables?

File: src/Request/Request.cpp [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+void Request::validateBody() {
+  size_t contentLength = static_cast<size_t>(_header.contentLength);
+  if (contentLength != _body.size()) {
+    throw RequestValidationException();
+  }
+}
+
+/*
+** --------------------------------- ACCESSOR ---------------------------------
```
Comment (koukunwlk) [inline src/Request/Request.cpp:148]: For functions that return a private field we can just call "Getters"

**Review by koukunwlk — CHANGES_REQUESTED** [review]

File: src/Request/Request.cpp (hunk 2 above)
Comment (dpiza) [reply src/Request/Request.cpp:148]: Getters and Setters are both Accessor functions. Plus this separator is added automatically by my class extension

File: include/Request/Request.hpp (hunk 1 above)
Comment (dpiza) [reply include/Request/Request.hpp:11]: They are used in the validateMethod function

Comment (koukunwlk) [reply include/Request/Request.hpp:11]: So... why don't use the native true and false?

File: src/Request/Request.cpp (hunk 2 above)
Comment (koukunwlk) [reply src/Request/Request.cpp:148]: IA

File: include/Request/Request.hpp (hunk 1 above)
Comment (dpiza) [reply include/Request/Request.hpp:11]: Because

**Review by koukunwlk — APPROVED** [review]
Watch out for suddenly recreating the libft in C++.
LGTM!

**Pull request merged**